### PR TITLE
DBZ-1376 Both ALGORTIHM and LOCK are allowed for INDEX

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1376

### CODE REVIEW

**Comment by jpechane** [comment]
Upstream issue https://github.com/antlr/grammars-v4/pull/1461/files

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 [hunk 1]
```diff
@@ -672,10 +672,9 @@ dropIndex
       uid ON tableName
       (
         ALGORITHM '='? algType=(DEFAULT | INPLACE | COPY)
-      )?
```
Comment (gunnarmorling) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4]: Isn't it a bit too lenient now? IOW, the one before looked right (in contrast to the other usage above), it allowed ALGORITHM and LOCK, no?

Comment (jpechane) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4]: But it prescribed order which is no strict

**Comment by Naros** [comment]
Rebased on master and merged.  Thanks @jpechane.

**Pull request closed without merging**